enigma/src/main/java/org/quiltmc/enigma/impl/source/vineflower/EnigmaTextTokenCollector.java

```diff
@@ -80,73 +108,273 @@ public void addTokensToIndex(SourceIndex index, UnaryOperator<Token> tokenProces
 		}
 	}
 
+	private void parseSource() {
+		ParserConfiguration config = new ParserConfiguration()
+				.setStoreTokens(true)
+				.setLanguageLevel(ParserConfiguration.LanguageLevel.RAW);
+
+		ParseResult<CompilationUnit> parseResult = new JavaParser(config).parse(this.content);
+		if (!parseResult.isSuccessful()) {
+			Logger.warn("Failed to parse source: {}", parseResult.getProblems());
+			return;
+		}
+
+		CompilationUnit unit = parseResult.getResult().get();
+		List<InitializerDeclaration> initializers = unit.findAll(InitializerDeclaration.class, InitializerDeclaration::isStatic);
+		for (InitializerDeclaration decl : initializers) {
+			Range range = decl.getRange().orElseThrow(() -> new IllegalStateException("No range for initializer"));
```

Comment:
In a brief look into javaparser I wasn't able to find a guarantee that ranges will be present.
If that's the case, I think an error should be loggerd and the declaration skipped if it has no range.